src-tauri/src/backend/notebook.rs

```diff
@@ -28,6 +28,20 @@ pub struct Notebook {
     pub cells: Vec<Cell>,
 }
 
+impl Notebook {
+    /// Loads a notebook from a JSON string.
+    pub fn load_from_str(s: &str) -> Result<Self, crate::Error> {
+        let notebook: Notebook = serde_json::from_str(s).map_err(crate::Error::from)?;
+        Ok(notebook)
+    }
+
+    /// Loads a notebook from a file path.
+    pub fn load_from_path(path: &str) -> Result<Self, crate::Error> {
+        let s = std::fs::read_to_string(path)?;
```

Comment:
Oh btw this is good to call from a sync function, but one gotcha you might not know about async rust is that this will prevent other async tasks from making progress while it's executing https://stackoverflow.com/questions/70599317/is-there-any-point-in-async-file-io